### PR TITLE
Introduce bruteforce settings

### PR DESCRIPTION
**Pull request by rullzer**
This adds an app with adds an admin settings. Lets the admin insert ip ranges (netmask) to exclude for throttling.

TODO:
- [x] Ignore ip ranges in throttler
- [x] Add JS tests
- [x] FIX CSS

@LukasReschke as discussed.

### CODE REVIEW

**Comment by LukasReschke** [comment]
Let's add this to https://github.com/nextcloud/server/blob/9fff4b017d0ab6196f182ba2bb2c4fb00187a11b/.drone.yml#L22-L34 as well :)

**Comment by rullzer** [comment]
@LukasReschke please have a look

**Comment by rullzer** [comment]
@MorrisJobke could you have a look at the crappy CSS? For some reason it wants text in order to show the trash icons

1. Go to admin page
2. Go to secrutiy tab
3. Add a subnet thingy
4. See bad icons
5. fix it 😉

**Comment by MorrisJobke** [comment]
I will have a look once I'm in the office :wink:

**Comment by MorrisJobke** [comment]
@rullzer The security page is empty 😢 

bildschirmfoto 2016-11-18 um 21 25 12

**Comment by rullzer** [comment]
Enable the app

**Comment by MorrisJobke** [comment]
Ah ... 🙈

**Comment by icewind1991** [comment]
@rullzer any update here?

**Comment by tx7** [comment]
Hello all,

Any ETA on this?

**Comment by MorrisJobke** [comment]
I rebased this, fixed the conflicts, improved the settings layout and it now looks like this:


bildschirmfoto 2017-03-13 um 19 43 48


hovered:

bildschirmfoto 2017-03-13 um 19 43 55

**Comment by MorrisJobke** [comment]
@nickvergessen @lukasreschke Ready for review \o/

**Comment by icewind1991** [comment]
@rullzer any reason to have this app in the server repo?

**Comment by MorrisJobke** [comment]
> @rullzer any reason to have this app in the server repo?

Because it's force enabled for now.

**Comment by rullzer** [comment]
@icewind1991 uhm. That was some time ago when I wrote this. But I would be fine with ripping it out... makes updating way easier ;)

**Comment by icewind1991** [comment]
@MorrisJobke we can still force enable apps that are in a seperate repo by bundling them for a release if needed.

**Comment by rullzer** [comment]
Let me rip this apart soonish.

**Comment by rullzer** [comment]
App in https://github.com/nextcloud/bruteforcesettings

**Comment by rullzer** [comment]
Ok lets get this in!

File: tests/karma.config.js [hunk 1]
```diff
@@ -118,6 +118,19 @@ module.exports = function(config) {
...
+				srcFiles: [
+					// need to enforce loading order...
+					'apps/bruteforcesettings/js/IPWhitelistModel.js',
+					'apps/bruteforcesettings/js/IPWhitelistCollection.js',
+					'apps/bruteforcesettings/js/IPWhitelistView.js',
+					'apps/bruteforcesettings/js/IPWhitelist.js',
+				],
+				testFiles: [
+					'apps/bruteforcesettings/tests/js/IPWhitelistSpec.js'
+				]
```
Comment (MorrisJobke) [inline tests/karma.config.js]: Shouldn't this be better executed in the apps repo? Otherwise the PR there aren't checked with the tests itself.

**Review by MorrisJobke — APPROVED** [review]
Beside the one comment this is fine 👍

File: tests/karma.config.js (hunk 1 above)
Comment (rullzer) [inline tests/karma.config.js]: Yes it should.

File: core/shipped.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
   "shippedApps": [
     "activity",
     "admin_audit",
+    "bruteforcesettings",
```
Comment (ChristophWurst) [inline core/shipped.json]: @rullzer what's that used for? I don't see an app `bruteforcesettings`

**Review by ChristophWurst — DISMISSED** [review]

File: core/shipped.json (hunk 2 above)
Comment (rullzer) [inline core/shipped.json]: Well it is just the list of shipped apps. I'm fine with removing that as well... if we decide to just ship it from the appstore...

Comment (ChristophWurst) [inline core/shipped.json]: Oh, didn't see your comment https://github.com/nextcloud/server/pull/2095#issuecomment-291008958. Now it makes sense!

**Review by LukasReschke — APPROVED** [review]

**Pull request merged**